docs/gemstash-multiple-sources.7.md

```diff
@@ -57,6 +61,53 @@ you want. The `/upstream` prefix tells Gemstash to use a gem source
 other than the default source. You can now bundle with the additional
 source.
 
+Notice that the third source doesn't need to be escaped.
+This is because the `https://` is used by default when no scheme is set,
+and the source URL does not contain any chars that need to be escaped.
+
+## Authentication with Multiple Sources
+
+You can use basic authentication or API keys on sources directly in Gemfile
+or using ENV variables on the Gemstash instance.
+
+Example `Gemfile`:
+```ruby
+# ./Gemfile
+require "cgi"
+source "http://localhost:9292"
+
+source "http://localhost:9292/upstream/#{CGI.escape("user:password@my.gem-source.local")}" do
+  gem "my-gem"
+end
+
+source "http://localhost:9292/upstream/#{CGI.escape("api_key@my-other.gem-source.local")}" do
+  gem "my-other-gem"
+end
+```
+
+If you set `GEMSTASH_<HOST>` ENV variable with your authentication information,
+you can omit it from the `Gemfile`:
+
+```ruby
+# ./Gemfile
+source "http://localhost:9292"
+
+source "http://localhost:9292/upstream/my.gem-source.local" do
+  gem "my-gem"
+end
+```
+
+And run the Gemstash with the credentials set in an ENV variable:
+
+```bash
+GEMSTASH_MY__GEM___SOURCE__LOCAL=user:password gemstash start --no-daemonize --config-file config.yml.erb
+```
+
+The name of the ENV variable is the uppercase version of the host name,
+with all `.` characters replaced with `__`, all `-` with `___` and a `GEMSTASH_` prefix.
```

Comment:
Minor: Perhaps we can link to Bundler's documentation about their host-in-environment variables? https://bundler.io/v2.4/man/bundle-config.1.html#CREDENTIALS-FOR-GEM-SOURCES